roles/etcd/templates/deb-etcd.initd.j2

```diff
@@ -12,15 +12,33 @@ set -a
 #   etcd is a distributed, consistent key-value store for shared configuration and service discovery
 ### END INIT INFO
 
-PATH=/sbin:/usr/sbin:/bin:/usr/bin
+
 DESC="etcd k/v store"
 NAME=etcd
+{% if etcd_deployment_type == "host" %}
 DAEMON={{ bin_dir }}/etcd
-{% if inventory_hostname in groups['etcd'] %}
+{% if is_etcd_master %}
 DAEMON_ARGS=""
 {% else %}
-DAEMON_ARGS="-proxy on"
+DAEMON_ARGS="--proxy on"
+{% endif %}
+{% else %}
+DAEMON={{ docker_bin_dir | default("/usr/bin") }}/docker
+{% if is_etcd_master %}
+DAEMON_ARGS='--env-file=/etc/etcd.env \
+--net=host \
+-v /usr/share/ca-certificates/:/etc/ssl/certs:ro \
+-v /var/lib/etcd:/var/lib/etcd:rw \
+--name={{ etcd_member_name | default("etcd-proxy") }} \
+{{ etcd_image_repo }}:{{ etcd_image_tag }} \
+{% if is_etcd_master %}
+--proxy off
+{% else %}
+--proxy on
+{% endif %}'
 {% endif %}
+
+
 SCRIPTNAME=/etc/init.d/$NAME
 DAEMON_USER=etcd
 STOP_SCHEDULE="${STOP_SCHEDULE:-QUIT/5/TERM/5/KILL/5}"
```

Comment:
What do you think to have 2 separate template files: 
- deb-etcd-host.initd.j2 
- deb-etcd-docker.initd.j2

And put the condition in the tasks. 

it will simplify the templates